stored_requests/backends/db_fetcher/fetcher.go

```diff
@@ -93,6 +97,47 @@ func (fetcher *dbFetcher) FetchRequests(ctx context.Context, requestIDs []string
 	return storedRequestData, storedImpData, errs
 }
 
+func (fetcher *dbFetcher) FetchResponses(ctx context.Context, ids []string) (data map[string]json.RawMessage, errs []error) {
+	if len(ids) < 1 {
+		return nil, nil
+	}
+
+	query := fetcher.responseQueryMaker(len(ids))
+	idInterfaces := make([]interface{}, len(ids))
+	for i := 0; i < len(ids); i++ {
+		idInterfaces[i] = ids[i]
+	}
+
+	rows, err := fetcher.db.QueryContext(ctx, query, idInterfaces...)
+	if err != nil {
+		return nil, []error{err}
+	}
+	defer func() {
+		if err := rows.Close(); err != nil {
+			glog.Errorf("error closing DB connection: %v", err)
+		}
+	}()
+
+	storedData := make(map[string]json.RawMessage, len(ids))
```

Comment:
You use `len(ids)` a few times in this function, lines 101, 105, 106, and here 121. Is the length of the `ids` array changing at all here? If no, you could just extract `len(ids)` to a variable, and use that variable instead, just to improve some efficiency here.